forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestJWTClaims.java

```diff
@@ -0,0 +1,211 @@
+        return value.get();
+    }
+
+
+    public Optional<Long> getOptionalLongValue(String claimName) throws DynamicClientRegistrationException {
+        try {
+            return Optional.ofNullable(this.claimsSet.getLongClaim(claimName));
+        } catch (ParseException pe){
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getOptionalLongValue() {}", errorString, pe);
+            throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+        }
+    }
+
+    public Date getRequiredDateValue(String claimName) throws DynamicClientRegistrationException {
+        Optional<Date> value = getOptionalDateValue(claimName);
+        if (value.isEmpty()){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                    "is required";
+            log.debug("getRequiredLongValue() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+        }
+        return value.get();
+    }
+
+    public Optional<Date> getOptionalDateValue(String claimName) throws DynamicClientRegistrationException {
+        return Optional.ofNullable((Date) this.claimsSet.getClaim(claimName));
+    }
+
+    public Double getRequiredDoubleClaim(String claimName) throws DynamicClientRegistrationException {
+        Optional<Double> value = getOptionalDoubleClaim(claimName);
+        if(value.isEmpty()){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claims is " +
+                    "required.";
+            log.debug("getRequiredDoubleClaim() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+        }
+        return value.get();
+    }
+
+    public Optional<Double> getOptionalDoubleClaim(String claimName) throws DynamicClientRegistrationException {
+        try {
+            return Optional.ofNullable(this.claimsSet.getDoubleClaim(claimName));
+        } catch (ParseException pe){
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getOptionalLongValue() {}", errorString, pe);
+            throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+        }
+    }
+
+
+    public Optional<List<String>> getOptionalStringListClaims(String claimName) throws DynamicClientRegistrationException {
+        try{
+            return Optional.ofNullable(claimsSet.getStringListClaim(claimName));
+        } catch (ParseException e){
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getStringListClaims() {}", errorString, e);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.

Reply:
After discussion, we worked out this should be log.info